File: problem_1/problem_1.go
package main

import (
	"fmt"
	"sort"
)

func diffMaxMin(numbers []int) int {
	if len(numbers) == 0 {
		return 0
	}
	sort.Ints(numbers)
	return numbers[len(numbers)-1] - numbers[0]
}

func main() {
	numbers := []int{100, 1, 50}
	fmt.Println(diffMaxMin(numbers))
}

File: programming/rust/src/problem_1.rs
fn diff_max_min(numbers: &[i32]) -> i32 {
    if numbers.is_empty() {
        0
    } else {
        numbers.iter().max().unwrap() - numbers.iter().min().unwrap()
    }
}

fn main() {
    let numbers = vec![100, 1, 50];
    println!("{}", diff_max_min(&numbers));
}
